src/java.base/share/classes/java/math/BigInteger.java

```diff
@@ -2640,8 +2638,8 @@ public BigInteger pow(int exponent) {
                     ? new BigInteger(result << bitsToShift, newSign)
                     : new BigInteger(result, newSign).shiftLeft(bitsToShift);
         }
-
-        if ((bitLength() - 1L) * exponent >= Integer.MAX_VALUE) {
+        // (this.abs().bitLength() - 1L) * exponent + 1L > Integer.MAX_VALUE
+        if (scaleFactor + bitsToShift - exponent >= Integer.MAX_VALUE) {
```

Comment:
If this is overflow sensitive code, should this avoid adding on the left hand side?

```suggestion
        if (scaleFactor - exponent >= Integer.MAX_VALUE - bitsToShift) {
```

Reply:
I think we are fine here, `scalreFactor` is already `long`

Reply:
> If this is overflow sensitive code, should this avoid adding on the left hand side?

`scaleFactor` is a `long`, so `scaleFactor + bitsToShift - exponent == (bitLength() - 1L) * exponent`.